packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -376,26 +565,31 @@ const Combobox = forwardRef<HTMLDivElement, ComboboxProps>(
       [renderNoResults]
     );
 
-    const noMatchingOptions = !!value?.length && !matchingOptions.size && (
+    const noMatchingOptions = !!inputValue && !matchingOptions.size && (
       <NoMatchingOptions />
     );
 
     const comboboxListbox = (
+      // eslint-disable-next-line
+      // @ts-ignore
```

Comment:
Nit: If needed, this is the better way to indicate we're expecting an issue rather than just outright ignoring it.

```suggestion
      // @ts-expect-error
```

Reply:
Updated